### PR TITLE
Delete and resize lvm thin pools and volumes

### PR DESCRIPTION
**Pull request by joseivanlopez**
PBI: https://trello.com/c/2F4R7E4p/207-1-lvm-expert-partitioner-delete-a-thin-provisioned-pool-and-volume

PBI: https://trello.com/c/6msuy7lb/206-2-lvm-expert-partitioner-resize-a-thin-provisioned-pool

### CODE REVIEW

File: test/y2partitioner/actions/delete_lvm_lv_test.rb [hunk 1]
```diff
@@ -32,43 +32,80 @@
...
     end
 
-    let(:device_name) { "/dev/vg1/lv1" }
-
     let(:accept) { nil }
 
-    it "shows a confirm message" do
-      expect(Yast::Popup).to receive(:YesNo)
-      subject.run
+    context "when the logical volume is not an used thin pool" do
```
Comment (shundhammer) [inline test/y2partitioner/actions/delete_lvm_lv_test.rb]: NP: "a used thin pool"

File: test/y2partitioner/actions/delete_lvm_lv_test.rb [hunk 2]
```diff
@@ -32,43 +32,80 @@
...
+
+      it "shows a confirmation message with the device name" do
+        expect(Yast::Popup).to receive(:YesNo) do |string|
+          expect(string).to include(device_name)
+        end
+        subject.run
+      end
+    end
+
+    context "when the logical volume is an used thin pool" do
```
Comment (shundhammer) [inline test/y2partitioner/actions/delete_lvm_lv_test.rb]: NP: "a used thin pool" (again)

File: test/y2partitioner/actions/delete_lvm_lv_test.rb [hunk 3]
```diff
@@ -32,43 +32,80 @@
...
 
     context "when the confirm message is not accepted" do
       let(:accept) { false }
 
+      let(:device_name) { "/dev/vg1/lv1" }
+
       it "does not delete the logical volume" do
         subject.run
-        expect(Y2Storage::BlkDevice.find_by_name(device_graph, device_name)).to_not be_nil
+        expect(device.exists_in_devicegraph?(current_graph)).to eq(true)
```
Comment (shundhammer) [inline test/y2partitioner/actions/delete_lvm_lv_test.rb]: 👍 much better readable

File: test/y2partitioner/actions/delete_lvm_lv_test.rb [hunk 4]
```diff
@@ -32,43 +32,80 @@
...
       end
 
       it "returns :back" do
         expect(subject.run).to eq(:back)
       end
+
+      context "and the logical volume is an used thin pool" do
+        let(:device_name) { "/dev/vg1/pool1" }
```
Comment (shundhammer) [inline test/y2partitioner/actions/delete_lvm_lv_test.rb]: NP (again): "a used thin pool"

File: test/y2partitioner/actions/delete_lvm_lv_test.rb [hunk 5]
```diff
@@ -79,6 +116,18 @@
       it "returns :finish" do
         expect(subject.run).to eq(:finish)
       end
+
+      context "and the logical volume is an used thin pool" do
```
Comment (shundhammer) [inline test/y2partitioner/actions/delete_lvm_lv_test.rb]: NP (again): "a used thin pool"

**Comment by shundhammer** [comment]
Completely LGTM now.

**Comment by joseivanlopez** [comment]
Thanks!

**Pull request merged**